neqo-transport/src/lib.rs

```diff
@@ -95,6 +99,7 @@ pub enum Error {
     VersionNegotiation,
     WrongRole,
     KeysDiscarded,
+    NoValidProtocolVersion,
```

Comment:
If you go with the enum, this might not be needed.

Reply:
Still needed?

Reply:
I think we still want it for if the `TryFrom<Version>` fails.